frame/proxy/src/lib.rs

```diff
@@ -166,16 +167,20 @@ decl_module! {
 			call: Box<<T as Trait>::Call>
 		) {
 			let who = ensure_signed(origin)?;
-			ensure!(T::IsCallable::filter(&call), Error::<T>::Uncallable);
 			let (_, proxy_type) = Proxies::<T>::get(&real).0.into_iter()
 				.find(|x| &x.0 == &who && force_proxy_type.as_ref().map_or(true, |y| &x.1 == y))
 				.ok_or(Error::<T>::NotProxy)?;
-			match call.is_sub_type() {
-				Some(Call::add_proxy(_, ref pt)) | Some(Call::remove_proxy(_, ref pt)) =>
-					ensure!(pt.is_no_more_permissive(&proxy_type), Error::<T>::NoPermission),
-				_ => (),
-			}
-			ensure!(proxy_type.filter(&call), Error::<T>::Unproxyable);
+
+			let _clear_guard = ClearFilterGuard::<T::IsCallable, <T as Trait>::Call>::new();
```

Comment:
Why do we need this?

Reply:
because we're now executing as a freshly authenticated new account, so the previous call restrictions no longer apply.